File: src/ChatApp/Services/Message/ChatApp.Message/Models/Message.cs
using System;
using System.Collections.Generic;

namespace ChatApp.Message.Models;

public sealed partial class Message
{
    public int Id { get; set; }

    public string Content { get; set; } = null!;

    public DateTime CreatedAt { get; set; }

    public DateTime? PinnedAt { get; set; }

    public int? RoomId { get; set; }

    public int? SenderId { get; set; }

    public DateTime? DeletedAt { get; set; }

    public int? DeletedBy { get; set; }

    public bool IsDeleted { get; set; }

    public User? DeletedByNavigation { get; set; }

    public ICollection<File> Files { get; set; } = new List<File>();

    public ICollection<MessageStatus> MessageStatuses { get; set; } = new List<MessageStatus>();

    public Room? Room { get; set; }

    public User? Sender { get; set; }
}

File: src/ChatApp/Webs/chatapp/src/app/chat/(types)/message.ts
export interface MessageDto {
  id: number;
  content: string | null;
  roomId: number;
  senderId: number;
  senderName: string;
  createdAt: string;
  status: MessageStatus;
  files?: FileDto[];
  isDeleted: boolean;
}

export interface FileDto {
  id: number;
  name: string;
  url: string;
  createdAt: string;
  type: string;
}

export type MessageStatus = 'Sending' | 'Sent' | 'Failed' | 'Delivered' | 'Seen';

export interface MessageStatusUpdate {
  messageId: number;
  status: MessageStatus;
  error?: string;
}

export interface PinnedMessage {
  id: number;
  roomId: number;
  senderId: number;
  senderName: string;
  content: string;
  files: FileDto[];
  pinnedAt: string;
  isPinned: boolean;
}

export interface MessageDeletedDto {
  messageId: number;
  roomId: number;
  deletedBy: number;
  deletedAt: string;
}